solidity/DieselPrice.sol

```diff
@@ -4,31 +4,30 @@
     This contract keeps in storage a reference
     to the Diesel Price in USD
 */
-pragma solidity ^0.4.0;
-
-import "github.com/oraclize/ethereum-api/oraclizeAPI.sol";
-
+pragma solidity ^0.4.25;
+import "https://raw.githubusercontent.com/oraclize/ethereum-api/master/oraclizeAPI_0.4.25.sol";
 
 contract DieselPrice is usingOraclize {
 
-    uint public dieselPriceUSD;
+    uint256 public dieselPriceUSD;
 
-    event NewOraclizeQuery(string description);
-    event NewDieselPrice(string price);
+    event NewOraclizeQuery(string _description);
+    event NewDieselPrice(string _price);
 
-    function DieselPrice() public {
+    constructor() public payable {
         update(); // first check at contract creation
     }
 
-    function __callback(bytes32 myid, string result) public {
+    function __callback(bytes32 _myid, string _result) public {
         if (msg.sender != oraclize_cbAddress()) revert();
```

Comment:
```suggestion
         if (msg.sender != oraclize_cbAddress()) revert();
        require(msg.sender == oraclize_cbAddress());
```

Could even add in a reason string - thoughts?

Reply:
That's a good idea! The only minor caveat I can think of is the gas price for reason strings; perhaps a note to suggest removing the strings on deployment to production?